src/transformers/models/luke/modeling_luke.py

```diff
@@ -0,0 +1,1362 @@
+
+        context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
+        context_layer = context_layer.view(*new_context_layer_shape)
+
+        output_word_hidden_states = context_layer[:, :word_size, :]
+        if entity_hidden_states is None:
+            output_entity_hidden_states = None
+        else:
+            output_entity_hidden_states = context_layer[:, word_size:, :]
+
+        if output_attentions:
+            outputs = (output_word_hidden_states, output_entity_hidden_states, attention_probs)
+        else:
+            outputs = (output_word_hidden_states, output_entity_hidden_states)
+
+        return outputs
+
+
+# Copied from transformers.models.bert.modeling_bert.BertSelfOutput
+class LukeSelfOutput(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, hidden_states, input_tensor):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class LukeAttention(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.self = LukeSelfAttention(config)
+        self.output = LukeSelfOutput(config)
+        self.pruned_heads = set()
+
+    def prune_heads(self, heads):
+        raise NotImplementedError("LUKE does not support the pruning of attention heads")
+
+    def forward(
+        self,
+        word_hidden_states,
+        entity_hidden_states,
+        attention_mask=None,
+        head_mask=None,
+        output_attentions=False,
+    ):
+        word_size = word_hidden_states.size(1)
+        self_outputs = self.self(
+            word_hidden_states,
+            entity_hidden_states,
+            attention_mask,
+            head_mask,
+            output_attentions,
+        )
+        if entity_hidden_states is None:
+            concat_self_outputs = self_outputs[0]
+            concat_hidden_states = word_hidden_states
+        else:
+            concat_self_outputs = torch.cat(self_outputs[:2], dim=1)
+            concat_hidden_states = torch.cat([word_hidden_states, entity_hidden_states], dim=1)
+
+        attention_output = self.output(concat_self_outputs, concat_hidden_states)
+
+        word_attention_output = attention_output[:, :word_size, :]
+        if entity_hidden_states is None:
+            entity_attention_output = None
+        else:
+            entity_attention_output = attention_output[:, word_size:, :]
+
+        outputs = (word_attention_output, entity_attention_output) + self_outputs[
+            2:
+        ]  # add attentions if we output them
```

Comment:
put the comment on the line above and make this one line

Reply:
Please apply this suggestion.

Reply:
Missed this accidentally, done.